code/modules/projectiles/gun.dm

```diff
@@ -63,6 +63,7 @@
 	var/scoped_accuracy = null
 	var/list/burst_accuracy = list(0) //allows for different accuracies for each shot in a burst. Applied on top of accuracy
 	var/list/dispersion = list(0)
+	var/is_charging = 0 //special var for sanity checks in the three guns that currently use charging as a special_check

```

Comment:
Ultimately I'd like to see charging guns used more, but I guess restricting it to energy/vaurca also works.